pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -127,35 +136,32 @@ def on_validation_end(self, trainer, pl_module):
             return
         if self.epochs_since_last_check >= self.period:
             self.epochs_since_last_check = 0
-            filepath = f'{self.filepath}/{self.prefix}_ckpt_epoch_{epoch}.ckpt'
+            filepath = os.path.join(self.dirpath, f'{self.prefix}_ckpt_epoch_{epoch}.ckpt')
             version_cnt = 0
             while os.path.isfile(filepath):
                 # this epoch called before
-                filepath = f'{self.filepath}/{self.prefix}_ckpt_epoch_{epoch}_v{version_cnt}.ckpt'
+                filepath = os.path.join(self.dirpath, f'{self.prefix}_ckpt_epoch_{epoch}_v{version_cnt}.ckpt')
                 version_cnt += 1
 
             if self.save_top_k != -1:
                 current = logs.get(self.monitor)
 
                 if current is None:
-                    warnings.warn(
-                        f'Can save best model only with {self.monitor} available,'
-                        ' skipping.', RuntimeWarning)
+                    warnings.warn(f'Can save best model only with {self.monitor} available,'
+                                  ' skipping.', RuntimeWarning)
                 else:
                     if self.check_monitor_top_k(current):
                         self._do_check_save(filepath, current, epoch)
                     else:
                         if self.verbose > 0:
-                            log.info(
-                                f'\nEpoch {epoch:05d}: {self.monitor}'
-                                f' was not in top {self.save_top_k}')
+                            log.info('Epoch %05d: %s was not in top %i', epoch, self.monitor, self.save_top_k)
 
             else:
                 if self.verbose > 0:
-                    log.info(f'\nEpoch {epoch:05d}: saving model to {filepath}')
+                    log.info('Epoch %05d: saving model to %s', epoch, filepath)
                 self._save_model(filepath)
 
-    def _do_check_save(self, filepath, current, epoch):
+    def _do_check_save(self, filepath: str, current, epoch):
```

Comment:
i like the addition of type hints, can we do it for all of the args though? (ie. `current` and `epoch`)